WeakAuras/Prototypes.lua

```diff
@@ -2191,11 +2191,20 @@ Private.event_prototypes = {
       ]=]
       if WeakAuras.IsRetail() then
         ret = ret .. [=[
-          local _, _, _, _, _, _, friendTextLevel = GetFriendshipReputation(factionID)
-          if friendTextLevel then
-            standing = friendTextLevel
+          if factionID then
+            local _, _, _, _, _, _, friendTextLevel = GetFriendshipReputation(factionID)
+            if friendTextLevel then
+              standing = friendTextLevel
+            end
+            local friendshipRank, friendshipMaxRank = GetFriendshipReputationRanks(factionID)
+            local currentValue, threshold, rewardQuestID, hasRewardPending, tooLowLevelForParagon  = C_Reputation.GetFactionParagonInfo(factionID)
+            if not tooLowLevelForParagon and currentValue and threshold then
+              local level = math.floor(currentValue/threshold)-(hasRewardPending and 1 or 0)
```

Comment:
Is that based on how blizzard does it?

Reply:
I used example from wiki https://wowpedia.fandom.com/wiki/API_C_Reputation.GetFactionParagonInfo but indeed blizzard code is better, changed
Also changed friendship reputation values like blizzard does